modules/core/src/directx.cpp

```diff
@@ -1397,6 +1397,33 @@ void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst
 #endif
 }
 
+#ifdef HAVE_OPENCL
+void convertFromD3D11Texture2DtoCLImage(ID3D11Texture2D* pD3D11Texture2D, cl_mem* clImage)
+{
+    CV_UNUSED(pD3D11Texture2D); CV_UNUSED(clImage);
+#if !defined(HAVE_DIRECTX)
+    NO_DIRECTX_SUPPORT_ERROR;
+#elif !defined(HAVE_OPENCL)
+    NO_OPENCL_SUPPORT_ERROR;
+#else
+    ID3D11Device *ppDevice = nullptr;
+    pD3D11Texture2D->GetDevice(&ppDevice);
+    cv::ocl::Context ctx = ocl::initializeContextFromD3D11Device(ppDevice);
+
+    OpenCL_D3D11* impl = ctx.getUserContext<OpenCL_D3D11>().get();
+    cl_context context = (cl_context)ctx.ptr();
+    if (nullptr == impl)
+        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initilized without DirectX interoperability");
+
+    cl_int status = 0;
+    *clImage = impl->clCreateFromD3D11Texture2DKHR(context, CL_MEM_WRITE_ONLY, pD3D11Texture2D, 0, &status);
```

Comment:
`CL_MEM_WRITE_ONLY` means you use texture as write-only  and each read-operation would return undefined behavior or forbidden (depends from implementation)

Reply:
pD3D11Texture2D, **0,**  
must be configurable because  textures may be represented as texture-array with subresource != 0

CL_MEM_WRITE_ONLY - may be function arguments  as well

Reply:
CL_MEM_READ_WRITE now